pkg/controllers/user/logging/utils/templatewrap.go

```diff
@@ -304,3 +342,81 @@ func parseEndpoint(endpoint string) (host string, scheme string, err error) {
 
 	return u.Host, u.Scheme, nil
 }
+
+func ValidateCustomTargetContent(content string, enableEscaped bool) (string, error) {
+	r := strings.NewReader(content)
+	buf := bufio.NewReader(r)
+	var newContent bytes.Buffer
+	for {
+		l, err := buf.ReadString('\n')
+		if l != "" {
+			line := strings.TrimSpace(l)
+			lineArray := strings.Split(line, " ")
+			if len(lineArray) == 0 {
+				continue
+			}
+
+			key := lineArray[0]
+			if err := filterFluentdTags(key); err != nil {
+				return "", err
+			}
+
+			if enableEscaped {
+				value := strings.TrimSpace(strings.TrimPrefix(line, key))
+				newValue := escapeString(value)
+				newContent.WriteString(key)
+				newContent.WriteString(" ")
+				newContent.WriteString(newValue)
+				newContent.WriteString("\n\t")
+			}
+		}
+
+		if err == io.EOF {
+			return newContent.String(), nil
+		}
+	}
+}
+
+func ValidateCustomTags(tags map[string]string, enableEscaped bool) (map[string]string, error) {
+	newCustomTags := make(map[string]string)
```

Comment:
You really don't need to create a new map to return. Just modify the content in `tags` map

Reply:
because tags map from the caller configureSyncer, the caller not supposed this value will be updated, so this value is no from a deep copy obj, is the original one, so I don't want to update the original one.